src/pretix/presale/views/cart.py

```diff
@@ -395,6 +395,10 @@ def dispatch(self, request, *args, **kwargs):
                     err = error_messages['voucher_redeemed']
                 if self.voucher.valid_until is not None and self.voucher.valid_until < now():
                     err = error_messages['voucher_expired']
+                if self.voucher.item is not None:
+                    if self.voucher.item.available_until is not None and (
+                            self.voucher.item.available_until < now()):
+                        err = error_messages['ended']
```

Comment:
I believe this message is misleading, we should rather create a new message that says
"Your voucher is valid for a product that is currently not for sale."